src/duration.rs

```diff
@@ -560,6 +561,13 @@ pub mod units {
                 }
             }
 
+            #[cfg(feature = "defmt")]
+            impl<T: TimeInt + defmt::Format> defmt::Format for $name<T> {
+                fn format(&self, fmt: defmt::Formatter) {
+                    defmt::write!(fmt, "{}", self.0)
```

Comment:
This will not produce the correct unit when printed.

Reply:
I.e. it should probably be added to the macro so `Milliseconds` prints as `xy ms`

Reply:
I made it so that it matches the current `core::fmt::Display` impl.

https://github.com/FluenTech/embedded-time/blob/59941050b645f31e963b5d7ffb0c0abac2fbb94a/src/duration.rs#L557-L569

I could change it for `core::fmt::Display` as well, but that would be a breaking change.

Reply:
Hmm true, what do you think here @PTaylor-us ?
I'd consider the current `fmt::Display` impl as a bug as it is now then.

Reply:
It looks like the core trait adds the unit abbreviation without the space ("1234ms"). I think it would be good to replicate that behavior. Thoughts?

https://doc.rust-lang.org/src/core/time.rs.html#1033-1154